RescueRS/Infrastructure/Repositories/RescueRepository.cs

```diff
@@ -22,30 +23,16 @@ public async Task InsertOrUpdate(RescueEntity rescue)
         await db.SaveChangesAsync();
     }
 
-    public async Task<IEnumerable<RescueEntity>> GetPendingRescues()
-    {
-        DateTimeOffset? lastDate = await this.db.Rescues
-                        .Where(x => x.RescueId == (Guid?)this.pagination.cursor)
-                        .Select(x => (DateTimeOffset?)x.RequestDateTime)
-                        .FirstOrDefaultAsync();
-
-        return await db.Rescues
-            .Where(x => !x.Rescued)
-            .OrderByDescending(x => x.RequestDateTime)
-            .ApplyPagination(this.pagination, x => lastDate == null || x.RequestDateTime < lastDate)
-            .ToListAsync();
-    }
-
-    public async Task<IEnumerable<RescueEntity>> GetPendingRescuesByProximity(double latitude, double longitude)
+    public async Task<IEnumerable<RescueEntity>> GetRescuesByProximityWithStatus(RescueStatusEnum status, double latitude, double longitude)
     {
         RescueEntity? lastRescue = await this.db.Rescues
                         .Where(x => x.RescueId == (Guid?)this.pagination.cursor)
                         .FirstOrDefaultAsync();
         double? lastDistance = lastRescue?.GetDistance(latitude, longitude);
 
         var rescues = await db.Rescues
```

Comment:
This logic is kind of strange. We are ordering twice. Will be marked as a technical debt and disscussed further more.